swan-lake/learn-the-platform/run-in-the-cloud/function-as-a-service/azure-functions.md

```diff
@@ -31,352 +26,137 @@ The following Azure Functions triggers and bindings are currently supported in B
+## Build the function
 
-Now, the deployed Azure Function can be tested by invoking it using an HTTP client such as cURL. 
+The Azure Functions functionality is implemented as a compiler extension. Thus, artifact generation happens automatically when you build a Ballerina module. 
 
-```bash
-$ curl --header "Content-Type: application/json" \
-  --request POST \
-  --data '{"name":"Jack","age":21}' \
-  "https://<function_app_name>.azurewebsites.net/queue"
-Jack Added to the Queue!
-```
-
-Refresh the queue page in the portal and view the added entry.
-
-### Cosmos DB trigger -> queue output
-
-The following Ballerina code gives an example of using a Cosmos DB trigger to invoke the function and a queue output binding to write an entry to a queue.
-
-Before writing and deploying the code, create a Cosmos DB and a queue to make use of those services later.
-1. You can reuse the queue you created in the above <a href="/learn/run-in-the-cloud/function-as-a-service/azure-functions/#http-trigger---queue-output" target="_blank">`HTTP trigger` -> `Queue output`</a> sample.
-2. Create an <a href="https://portal.azure.com/#create/Microsoft.DocumentDB" target="_blank">Azure Cosmos DB account</a> and select Cosmos DB Core.
-3. Once the database is created, go to the **Data Explorer**, and select **Create Container**.
-4. Enter `db1` as Database ID and `c1` as the collection ID, and click **Ok**.
-
-**Note:** If you want to change these values, make sure to change them in the code as well.
-5. Go to the **Keys** tab of the Cosmos DB page.
-6. Copy the value of the `PRIMARY CONNECTION STRING`.
-7. Click the **Configuration** tab on the Function app page.
-8. Select **New Application Setting**, and paste the data you copied above as the value. For the key, use the value of the `connectionStringSetting` key and save.
-
-Example application setting is as follows.
-```
-Name - `CosmosDBConnection`
-Value - `AccountEndpoint=https://db-cosmos.documents.azure.com:443/;AccountKey=12345asda;`
-```
-
-Now, as all the infrastructure required are up and running and configured, start building and deploying the Azure function.
-
-```ballerina
-import ballerina/log;
-import ballerinax/azure_functions as af;
+## Deploy the function
 
-public type DBEntry record {
-    string id;
-    string name;
-};
+The created function app name should be provided to the placeholders shown in the above-generated usage instructions from the compiler. 
 
-@af:CosmosDBTrigger {connectionStringSetting: "CosmosDBConnection", databaseName: "db1", collectionName: "c1"}
-listener af:CosmosDBListener cosmosEp = new ();
+## Invoke the function
 
-service "cosmos" on cosmosEp {
-    remote function onUpdate(DBEntry[] entries) returns @af:QueueOutput {queueName: "people"} string {
-        string name = entries[0].name;
-        log:printInfo(entries.toJsonString());
-        return "Hello, " + name;
-    }
-}
+The deployed Azure Function can be tested by invoking it using an HTTP client such as cURL.
```

Comment:
Same as AWS lambda, each function can have different triggers. Ex - if the trigger is queue, we can't use curl to invoke it. Therefore, we need to decide if we are keeping this section.

Reply:
Removed as invoking will be demonstrated in the BBEs.